### PR TITLE
dbeaver/dbeaver#18348 add legacy ssh-rsa

### LINKED ISSUE #18348: SSH MFA Auth with interactive password prompt broken since 22.2.1
### Description

Since 22.2.1 release got following error while trying to connect DB over SSH Tunnel & Jump server with MFA enabled.
Error details:
```
Can't initialize tunnel
Cannot establish tunnel to 10.0.2.10:22
Cannot establish tunnel to 10.0.2.10:22
SSH_MSG_DISCONNECT: 2 Too many authentication failures 
SSH_MSG_DISCONNECT: 2 Too many authentication failures 

```
In 22.2.0 Password prompt appeared when trying to connect (expected behavior, as password was not saved in Dbeaver Connection Preferences/SSH Tab), and then connection successfully established.

Since 22.2.1 SSH password prompt not showing and connection automatically fails

image

### DBeaver Version

22.2.1-22.2.5

### Operating System

macOS BigSur 11.4

### Database and driver

SSH Driver used: JSch

### Steps to reproduce

1. Enable MFA (key+password) on linux host (JumpHost for example)

```
#/etc/ssh/sshd_config:
AuthenticationMethods publickey,password
```
2. Add ssh public key to user .ssh folder
3. Do not save passphrase in SSH Tunnel settings
4. Try to Test tunnel configuration by pressing "Test tunnel configuration" key

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Destrolaric**
Closes dbeaver/dbeaver#18348

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.net.ssh.jsch/src/org/jkiss/dbeaver/model/net/ssh/SSHImplementationJsch.java [hunk 1]
```diff
@@ -58,6 +58,11 @@ protected synchronized void setupTunnel(@NotNull DBRProgressMonitor monitor, @No
         if (jsch == null) {
             jsch = new JSch();
             JSch.setLogger(new JschLoggerProxy());
+            if (JSch.getConfig("PubkeyAcceptedAlgorithms") != null && !JSch.getConfig("PubkeyAcceptedAlgorithms").contains(
+                "ssh-rsa")) {
+                JSch.setConfig("server_host_key", JSch.getConfig("server_host_key") + ",ssh-rsa");
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.net.ssh.jsch/src/org/jkiss/dbeaver/model/net/ssh/SSHImplementationJsch.java]: Check that server_host_key present in confif and it is not empty.

File: plugins/org.jkiss.dbeaver.net.ssh.jsch/src/org/jkiss/dbeaver/model/net/ssh/SSHImplementationJsch.java [hunk 2]
```diff
@@ -58,6 +58,11 @@ protected synchronized void setupTunnel(@NotNull DBRProgressMonitor monitor, @No
         if (jsch == null) {
             jsch = new JSch();
             JSch.setLogger(new JschLoggerProxy());
+            if (JSch.getConfig("PubkeyAcceptedAlgorithms") != null && !JSch.getConfig("PubkeyAcceptedAlgorithms").contains(
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.net.ssh.jsch/src/org/jkiss/dbeaver/model/net/ssh/SSHImplementationJsch.java]: Use variable

**Review by serge-rider — CHANGES_REQUESTED** [review]

File: plugins/org.jkiss.dbeaver.net.ssh.jsch/src/org/jkiss/dbeaver/model/net/ssh/SSHImplementationJsch.java [hunk 3]
```diff
@@ -58,6 +58,13 @@ protected synchronized void setupTunnel(@NotNull DBRProgressMonitor monitor, @No
         if (jsch == null) {
             jsch = new JSch();
             JSch.setLogger(new JschLoggerProxy());
+            String serverHostKeys = JSch.getConfig("server_host_key");
+            String pubkeyAcceptedAlgorithms = JSch.getConfig("PubkeyAcceptedAlgorithms");
+            if (!CommonUtils.isEmpty(pubkeyAcceptedAlgorithms) && !CommonUtils.isEmpty(serverHostKeys) && !pubkeyAcceptedAlgorithms.contains(
+                "ssh-rsa")) {
+                JSch.setConfig("server_host_key", serverHostKeys + ",ssh-rsa");
+                JSch.setConfig("PubkeyAcceptedAlgorithms", pubkeyAcceptedAlgorithms + ",ssh-rsa");
```
Comment (E1izabeth) [inline plugins/org.jkiss.dbeaver.net.ssh.jsch/src/org/jkiss/dbeaver/model/net/ssh/SSHImplementationJsch.java:66]: You use `"server_host_key"` and `"PubkeyAcceptedAlgorithms"` at least twice here. I think it's better to make a constant and use it instead.

Also, I'd make `ssh-rsa` to be a constant.

**Comment by ShadelessFox** [comment]
It is worth enabling the algorithm on the library side. Too much hassle

**Pull request closed without merging**